<launch>
  <param name="/use_sim_time" value="true"/>

  <!-- Nav stack (copied from 2dnav_pr2.launch, with amcl and map server left out) -->
  <include file="$(find pr2_machine)/sim.machine" />
  <include file="$(find 2dnav_pr2)/config/base_odom_teleop.xml" />
  <include file="$(find 2dnav_pr2)/config/lasers_and_filters.xml" />
  <include file="$(find 2dnav_pr2)/config/ground_plane.xml" />  
  <include file="$(find 2dnav_pr2)/move_base/move_base.xml" />  

  <!-- Overrides of some navstack parameters -->
  <param name="move_base/base_global_planner" value="NavfnROSConstrained"/>  
  <rosparam file="$(find people_aware_nav)/launch/global_costmap_params.yaml" 
	    command="load" ns="move_base"/>



  <!-- Hallway related -->
    
    <param name="hallway_global_frame_id" value="odom_combined" />

    <!-- Hallway tracker -->
    <include file="$(find hallway_tracker)/launch/hallway_tracker.launch" />
    <param name="/hallway_tracker_node/p_max_hallway_width_m" type="double" value="5.5" />

    <!-- Various transforms required by exec -->
    <node pkg="people_aware_nav" type="person_pos_sender" name="person_pos_sender" output="screen" />

    <!-- Node that checks if person is on currently planned path -->
    <node pkg="people_aware_nav" type="is_person_on_path" name="is_person_on_path" output="screen" >
      <param name="person_radius" type="double" value="1.2"/>
    </node>



  <!-- <node pkg="pr2_mechanism_controllers" type="tuckarm.py" args="r"/> -->
  <!-- node pkg="people_aware_nav" type="hallway_move" output="screen" /> -->

</launch>
